resteasy-core/src/main/java/org/jboss/resteasy/core/providerfactory/ResteasyProviderFactoryImpl.java

```diff
@@ -1115,17 +1115,18 @@ public <T extends Throwable> ExceptionMapper<T> getExceptionMapperForClass(Class
    public <T> AsyncResponseProvider<T> getAsyncResponseProvider(Class<T> type)
    {
       Class asyncType = type;
-      AsyncResponseProvider<T> mapper = null;
-      while (mapper == null)
+      if (getAsyncResponseProviders() == null)

```

Comment:
Should the three calls to `getAsyncResponseProviders()` be stored in a local and re-used?